harness/determined/common/storage/base.py

```diff
@@ -87,7 +107,17 @@ def upload(self, src: Union[str, os.PathLike], dst: str) -> None:
         pass
 
     @abc.abstractmethod
-    def download(self, src: str, dst: Union[str, os.PathLike]) -> None:
+    def download(
+        self,
+        src: str,
+        dst: Union[str, os.PathLike],
+        selector: Optional[Selector] = None,
+    ) -> None:
+        """
+        `selector` should be a callable accepting a string parameter, ending in an os.sep if it is a
+        directory, and should return bool: True for files/directories that should be downloaded;
```

Comment:
"and should return ~bool:~ True for files/directories that should be downloaded or False otherwise"